packages/appkit/src/client.ts

```diff
@@ -114,8 +115,24 @@ export class AppKit {
     return ChainController.state.activeCaipNetwork?.id
   }
 
-  public switchNetwork(caipNetwork: CaipNetwork) {
-    return ChainController.switchActiveNetwork(caipNetwork)
+  public addNetwork(caipNetwork: CaipNetwork) {
+    ChainController.addCaipNetwork(caipNetwork)
+  }
+
+  public switchNetwork(appKitNetwork: AppKitNetwork | CaipNetwork) {
+    const network = this.caipNetworks.find(n => n.caipNetworkId === appKitNetwork.id)
+    const extendedNetwork =
+      network ||
+      CaipNetworksUtil.extendCaipNetwork(appKitNetwork, {
+        projectId: OptionsController.state.projectId,
+        customNetworkImageUrls: {}
+      })
+
+    if (!network) {
+      ChainController.addCaipNetwork(extendedNetwork)
+    }
```

Comment:
is this ok? should we allow non available networks to be switched to?

Reply:
If they provide non available network, we first need to add it to the state, then switch to it. Honestly, better to handle this on the tests as well